src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java

```diff
@@ -33,11 +33,19 @@
 
 	private JButton Play, Instruction, Exit, Resume, Easy, Medium, Hard, Back, Menu, Credit, PlayBGM, PauseBGM, HighScore, NextPage;
 	private JButton Character, CMenu, Kwhale, Shark;
-	private JFrame frame, instruct;
+
 
 //	private int type;
 //	private JTextArea text;
 //	private JPanel textpan;
+
+	public static JFrame frame, instruct;
+	public static int diffBG;
```

Comment:
Making this static on `Menu` isn't a great place to put it.  Can you pass this to the constructor of `FishAnimationEnvironment` instead?  Or - even better - can you make `FishAnimationEnvironment` draw the correct background based on the `difficulty` member variable it already has?

Reply:
I tried, but failed. If I put them in the constructor and remove them in Menu.java, the program will crash. I think it is better to leave it there.

Reply:
You don't need a `diffBG` variable at all.  Just decide what to draw based on the `difficulty` variable.

As a general rule, treat static variables with _almost_ as much suspicion as you treat global variables.  There's a small chance they're there for a good reason, but most of the time it's just poor design.

Reply:
get it. fixed